src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -0,0 +1,492 @@
+			for(FloatingTask t: masterList.getFloatingTasksList()){
+				if(t.getCompleted()){
+					rowNo++;
+					add(t);
+				}
+			}
+		}
+
+		return hasCompleted;
+	}
+
+	public void showSearchResults(){
+		pane.getChildren().clear();
+		rowNo = new Integer(0);
+
+		Label titleLabel = new Label("Search Results");
+		titleLabel.getStyleClass().add("header");
+		pane.add(titleLabel, 0, rowNo, NO_OF_COLUMNS, 1);
+
+		LocalDate firstDate = getFirstDate();
+
+		if(firstDate == null && masterList.getFloatingTasksList().isEmpty()){
+			Label emptyLabel = new Label("No results match your search term");
+			emptyLabel.getStyleClass().add("red-label");
+			pane.add(emptyLabel, 0, 1, NO_OF_COLUMNS, 1);
+		}
+		else if (firstDate != null) {
+			LocalDate tempDate = firstDate;
+
+			while(!tempDate.isAfter(getLastDate())){
+				addTaskEventsOnDate(tempDate);
+				tempDate = tempDate.plusDays(1);
+			}
+		}
+
+		for(FloatingTask task: masterList.getFloatingTasksList()){
+			add(task);
+		}
+	}
+
+	private int addTaskEventsOnDate(LocalDate date){
+		Label dateLabel = new Label(date.format(dateFmt));
+		dateLabel.getStyleClass().add("date-label");
+		pane.add(dateLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		int noOfEvents = addEventOnDate(date);
+		int noOfTasks = addDeadlineTaskOnDate(date);
+
+		rowNo += (noOfEvents+noOfTasks);
+
+		Label emptyLabel = new Label("");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if ((noOfEvents+noOfTasks) == 0){
+			pane.getChildren().remove(dateLabel);
+			pane.getChildren().remove(emptyLabel);
+		}
+
+		return noOfEvents + noOfTasks;
+	}
+
+	private int addTaskEventsOnDate(LocalDate date, LocalDate selectedDate){
+		Label dateLabel = new Label(date.format(dateFmt));
+		dateLabel.getStyleClass().add("date-label");
+		pane.add(dateLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		int noOfEvents = addEventOnDate(date);
+		int noOfTasks = addDeadlineTaskOnDate(date);
+
+		rowNo += (noOfEvents+noOfTasks);
+
+		Label emptyLabel = new Label("");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+
+		if ((noOfEvents+noOfTasks) == 0){
+			if(selectedDate.equals(date)){
+				pane.getChildren().remove(emptyLabel);
+				addNoEventTaskLabel(date);
+				pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+			}
+			else {
+				pane.getChildren().remove(dateLabel);
+				pane.getChildren().remove(emptyLabel);
+			}
+
+		}
+
+		return noOfEvents + noOfTasks;
+	}
+
+	private void addNoEventTaskLabel(LocalDate currentDate){
+
+		if(rowNo == -1){
+			Label selectedDateLabel = new Label(currentDate.format(dateFmt));
+			selectedDateLabel.getStyleClass().add("date-label");
+			pane.add(selectedDateLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+		}
+
+		Label emptyLabel = new Label("No events or deadline tasks on this day");
+		emptyLabel.getStyleClass().add("red-label");
+		pane.add(emptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+	}
+
+	private int addEventOnDate(LocalDate date){
+		int noOfEvents = 0;
+
+		for(Event event: masterList.getEventsList()){
```

Comment:
addEventOnDate is a one time use only for the Today Controller which finds only one day now.